### PR TITLE
Add ActiveModel::Errors methods.

### PR DESCRIPTION
**Pull request by connorshea**
And errors method on ActiveModel::Validations, plus some tests.

[`ActiveModel::Errors`](https://api.rubyonrails.org/classes/ActiveModel/Errors.html) is used to handle errors when validating models in Rails. This adds some of the most common methods for this :)

### CODE REVIEW

File: lib/activemodel/all/activemodel_test.rb [hunk 1]
```diff
@@ -77,3 +77,24 @@ module ActiveModelTest
   validates_numericality_of :width, greater_than: :minimum_weight, on: [:create, :update]
   validates_presence_of :first_name, on: [:create, :update]
 end
+
+class Person
```
Comment (ghiculescu) [inline lib/activemodel/all/activemodel_test.rb]: can you please add a test where this descends from `ActiveRecord::Base` too?

there were issues with the `ActiveModel::Validations` sigs where they didn't work properly in AR, which a test would have caught - see https://github.com/sorbet/sorbet-typed/pull/216/files#diff-52d2b7591f89a773def6fce9c1ca7ad7R80 for an example.

Comment (connorshea) [reply lib/activemodel/all/activemodel_test.rb]: You mean `ApplicationRecord`? I can, but it causes Person not to have a `new` method so I have to write the initializer like `person = T.let(T.unsafe(nil), Person)`

**Comment by DarkDimius** [comment]
@ghiculescu should we merge this?

**Comment by ghiculescu** [comment]
This is fine to merge, it just may not actually do what's expected without the extra test I suggested.

File: lib/activemodel/all/activemodel_test.rb (hunk 1 above)
Comment (ghiculescu) [reply lib/activemodel/all/activemodel_test.rb]: @connorshea did you also add this line? https://github.com/sorbet/sorbet-typed/pull/216/files#diff-52d2b7591f89a773def6fce9c1ca7ad7R39

Comment (connorshea) [reply lib/activemodel/all/activemodel_test.rb]: @ghiculescu that link doesn't actually refer to a specific line.

Comment (ghiculescu) [reply lib/activemodel/all/activemodel_test.rb]: Indeed it doesn't, and now I can't remember what I was trying to link to :(

Comment (ghiculescu) [reply lib/activemodel/all/activemodel_test.rb]: Oh right. It was this line: https://github.com/sorbet/sorbet-typed/blob/master/lib/activerecord/all/activerecord_test.rb#L39

**Pull request merged**